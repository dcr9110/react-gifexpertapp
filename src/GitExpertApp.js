import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GitExpertApp = ( {defaultCategories = ['Samurai X'] } ) => {

    const [categorias, setCategorias] = useState(defaultCategories);
   
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias={ setCategorias }/>
            <hr/>
            
            {
                categorias.map( category  => (
                    <GifGrid 
                        key={ category }
                        category={ category } 
                    />
                ))
            }
        </div>
    )
}
