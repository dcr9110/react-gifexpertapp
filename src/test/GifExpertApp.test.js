import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GitExpertApp } from '../GitExpertApp';

describe('Pruebas con el componente <GifExpertApp/>', () => {
    
    test('Tiene que mostrar correctamente el componente', () => {
        
        const wrapper = shallow(<GitExpertApp/>);
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GitExpertApp defaultCategories={categories}/> );

        expect(wrapper.find('GifGrid').length).toBe( categories.length );
    })
    

})
