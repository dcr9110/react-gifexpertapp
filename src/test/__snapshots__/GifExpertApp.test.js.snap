// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifExpertApp/> Tiene que mostrar correctamente el componente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    category="Samurai X"
    key="Samurai X"
  />
</div>
`;
